WordPress/Classes/Services/EditorSettingsService.swift

```diff
@@ -59,7 +59,7 @@ import Foundation
         let remoteEditor: EditorSettings.Mobile = isGutenbergEnabled ? .gutenberg : .aztec
 
         let service = EditorServiceRemote(wordPressComRestApi: api)
-        service.postDesignateMobileEditorForAllSites(remoteEditor, success: { response in
+        service.postDesignateMobileEditorForAllSites(remoteEditor, setOnlyIfEmpty: !overrideRemote, success: { response in
```

Comment:
There is an issue in the API implementation. The API checks for `setOnlyIfEmpty` presence, not for its value.  `setOnlyIfEmpty: false` doesn't work, we have to omit the parameter from the network call.

I guess, this must be changed: 
https://github.com/wordpress-mobile/WordPressKit-iOS/blob/a06182b86d3c8542c16ce38c765769e1d50b0d5f/WordPressKit/EditorServiceRemote.swift#L28 (don't set the `set_only_if_empty` parameter if `setOnlyIfEmpty` is false).

Reply:
Related FluxC PR: https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/1466

Reply:
Interesting! On my tests, it wasn't working before because `setOnlyIfEmpty` was true. I changed it to `false` and it started working properly 🤔 

I mean, the site that was set to `aztec` in the report card did change to `gutenberg`.

Reply:
I'll double check 👍

Reply:
> I mean, the site that was set to aztec in the report card did change to gutenberg

I double checked this and seems to be working. Was it another the issue? @maxme

Reply:
I tried again and it doesn't work with my test user:

Example with `setOnlyIfEmpty:false` and without:
```
$ curl 'https://public-api.wordpress.com/wpcom/v2/me/gutenberg/' -H 'Authorization: Bearer SECRET' -H 'Content-Type: application/json; charset=utf-8' --data 

'{"editor":"gutenberg","set_only_if_empty":"false","platform":"mobile"}'
{"59073674":"aztec","61784930":"aztec","64438643":"aztec","64486761":"aztec","67076078":"aztec","67127718":"aztec","70130817":"aztec","74198645":"aztec","74198675":"aztec","77334997":"aztec","97416322":"aztec","123950227":"aztec","124661775":"aztec","124665952":"aztec","143076330":"aztec","161993364":"aztec","162263259":"aztec","163024146":"aztec","163024199":"aztec","163024335":"aztec","163024524":"aztec","163218637":"aztec","163228098":"aztec","163662735":"aztec","123475559":"aztec"}

$ curl 'https://public-api.wordpress.com/wpcom/v2/me/gutenberg/' -H 'Authorization: Bearer SECRET' -H 'Content-Type: application/json; charset=utf-8' --data 

'{"editor":"gutenberg","platform":"mobile"}'
{"59073674":"gutenberg","61784930":"gutenberg","64438643":"gutenberg","64486761":"gutenberg","67076078":"gutenberg","67127718":"gutenberg","70130817":"gutenberg","74198645":"gutenberg","74198675":"gutenberg","77334997":"gutenberg","97416322":"gutenberg","123950227":"gutenberg","124661775":"gutenberg","124665952":"gutenberg","143076330":"gutenberg","161993364":"gutenberg","162263259":"gutenberg","163024146":"gutenberg","163024199":"gutenberg","163024335":"gutenberg","163024524":"gutenberg","163218637":"gutenberg","163228098":"gutenberg","163662735":"gutenberg","123475559":"gutenberg"}
```

I haven't checked the network code on iOS, can you sniff the network call when you set `setOnlyIfEmpty` to `false`?

Reply:
```json
{
	"set_only_if_empty": false,
	"platform": "mobile",
	"editor": "gutenberg"
}
```

There is a small difference: On your snippet you are sending a string `"false"`, on iOS we send a bool value `false`. Could you try to send `false` without the `"`?

Reply:
That's it 🤦‍♂